src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectMiddleware.cs

```diff
@@ -82,6 +82,11 @@ public class OpenIdConnectMiddleware : AuthenticationMiddleware<OpenIdConnectOpt
                 throw new ArgumentNullException(nameof(htmlEncoder));
             }
 
+            if (string.IsNullOrEmpty(Options.ClientId))
+            {
+                throw new ArgumentException(nameof(Options.ClientId));
```

Comment:
Wrong constructor. Use the two param one: https://msdn.microsoft.com/en-us/library/sxykka64(v=vs.110).aspx